### PR TITLE
✨(api) add tasks routes to trigger celery tasks

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

Adding `tasks` API routes to:
- trigger a task with a POST
- get the list of available tasks
- get the status of a task

### CODE REVIEW

File: src/mork/api/auth.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 api_key_header = APIKeyHeader(name="X-API-Key")
 
 
-def get_user(api_key_header: str = Security(api_key_header)):
```
Comment (lunika) [inline src/mork/api/auth.py:11]: You should already make test. Otherwise later it will be harder to do it.

File: src/mork/api/routes/tasks.py [hunk 2]
```diff
@@ -2,8 +2,40 @@
...
+
+    task = TaskResponse(id=result.task_id, status=TaskStatus.PENDING)
+    response.headers["location"] = router.url_path_for(
+        "get_task_status", **{"task_id": task.id}
+    )
+    return task
+
+
+@router.get("/")
+async def get_available_tasks() -> dict:
```
Comment (lunika) [inline src/mork/api/routes/tasks.py:31]: This is confusing. I expect to have all processing task in celery. IMO this should be an `OPTIONS` on the endpoint `/tasks`

File: src/mork/api/routes/tasks.py [hunk 3]
```diff
@@ -2,8 +2,40 @@
...
+from mork.worker.tasks import TASK_TYPE_TO_FUNC
 
 logger = logging.getLogger(__name__)
 
-router = APIRouter(prefix="/tasks")
+router = APIRouter(prefix="/tasks", dependencies=[Depends(authenticate_api_key)])
+
+
+@router.post("/", status_code=status.HTTP_202_ACCEPTED)
+async def create_task(task_in: TaskCreate, response: Response) -> TaskResponse:
```
Comment (lunika) [inline src/mork/api/routes/tasks.py:18]: Same here. Add the tests the same time you add new piece of code.

File: src/mork/worker/tasks.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+"""Celery tasks for sending emails to inactive users."""
```
Comment (quitterie-lcs) [inline src/mork/worker/tasks.py:1]: To be renamed as the deletion task is also defined in this file

File: tests/api/test_tasks.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    celery_result(task_id).state = "SUCCESS"
+
+    with patch("mork.api.tasks.AsyncResult", celery_result):
+        response = await http_client.get(
+            f"/tasks/status/{task_id}",
+            headers=auth_headers,
+        )
+        response_data = response.json()
+        assert response.status_code == 200
+        assert response_data
```
Comment (lunika) [inline tests/api/test_tasks.py:95]: You can probably assert that the response contains the state `SUCCESS` ?

Comment (wilbrdt) [reply tests/api/test_tasks.py]: Indeed good catch!

**Review by quitterie-lcs — APPROVED** [review]
LGTM

**Pull request merged**